### ISSUE TITLE
little suggestion on pad_token check in LlamaForSequenceClassification class

### ISSUE THREAD

**Issue by Alessa-mo** [huggingface/transformers#34055]
### System Info

When I use LlamaForSequenceClassification for training the official llama3.1 8b model on multi label classification task, as we know, llama model doesn't have a pad_token, we should set the pad_token.
Usually, we write 
`tokenizer.pad_token = tokenizer.eos_token` or `tokenizer.add_special_tokens({'pad_token': '[PAD]'})`
However, the code for check pad_token in the LlamaForSequenceClassification class is:
```
if self.config.pad_token_id is None and batch_size != 1:
            raise ValueError("Cannot handle batch sizes > 1 if no padding token is defined.")
```
It reads the model config file and then checks it. So I have to modify the config file instead of simply adding the pad_token set code in my code. Otherwise, it will raise the error. It's quite inconvenient. I think maybe it can be improved.

### Who can help?

@ArthurZucker

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

`tokenizer = PreTrainedTokenizerFast.from_pretrained("meta-llama/Meta-Llama-3.1-8B")`
`model = LlamaForSequenceClassification.from_pretrained("meta-llama/Meta-Llama-3.1-8B", num_labels=num_labels)`
`if tokenizer.pad_token is None:
    tokenizer.pad_token =  tokenizer.eos_token`
call any function which invovles the forward process, like training or infering.

### Expected behavior

don't need to change the config file to avoid these exception

**Comment by Alessa-mo** [comment]
I misunderstand the LlamaForSequnceClassification code

**Comment by ArthurZucker** [comment]
No worries!

**Comment by bilalghanem** [comment]
There is an issue with this check. I load llama tokenizer and I added the padding token:


```
tokenizer = AutoTokenizer.from_pretrained(model_params["MODEL"])
    if tokenizer.pad_token is None: 
        tokenizer.pad_token = tokenizer.eos_token
```


but then I get the error:

> File ".../decoder_to_encoder/venv/lib/python3.8/site-packages/transformers/models/llama/modeling_llama.py", line 1391, in forward                                                     │·····················
>     raise ValueError("Cannot handle batch sizes > 1 if no padding token is defined.")
>
